debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -112,23 +118,19 @@ public String getTable() {
         for (int i = 0; i < columnNames.size(); i++) {
             String columnName = columnNames.get(i).asString();
             String columnType = columnTypes.get(i).asString();
+            boolean columnOptional = columnOptionals != null ? columnOptionals.get(i).asBoolean() : false;
             Value rawValue = columnValues.get(i);
 
-            columns.add(new ReplicationMessage.Column() {
+            columns.add(new AbstractReplicationMessageColumn(columnName, columnType, columnOptional, true) {
 
                 @Override
                 public Object getValue(PgConnectionSupplier connection) {
-                    return Wal2JsonReplicationMessage.this.getValue(columnName, columnType, rawValue, connection);
+                    return Wal2JsonReplicationMessage.this.getValue(columnName, getTypeMetadata(), rawValue, connection);
                 }
 
                 @Override
-                public String getType() {
-                    return columnType;
-                }
-
-                @Override
-                public String getName() {
-                    return columnName;
+                protected int getOidType() {
+                    return 0;
```

Comment:
Why always 0?

Reply:
Thanks, something tells me wou should extend our test suite a little bit.